[SN-121]

jbkkkkk opened the pull request.

jbkkkkk committed:
[SN-121]
Создан проект для панели администрирования

jbkkkkk committed:
Merge remote-tracking branch 'origin/refactor/change-architecture' into features/add-admin-panel

jbkkkkk committed:
[SN-121]
Имплементирован функционал рассмотрения заявок на создание клуба

c23p1 commented on src/Infrastructure/Services/Queries/ClubCreationRequestsQueryService.cs:
```diff
@@ -0,0 +1,27 @@
+    }
+    
+    public async Task<ICollection<ClubCreationRequestDetailsDTO>> GetActiveRequestsAsync() 
+    => await _clubCreationRequestsDetails.Where(x => x.IsActive == true).ToListAsync();
```
Redundant "==" operator

c23p1 commented on src/Application/Interfaces/Services/IClubCreationRequestsService.cs:
```diff
@@ -0,0 +1,9 @@
+
+public interface IClubCreationRequestsService
+{
+    public Task<int> CreateRequest(ClubCreationRequestDTO clubCreationRequestDTO);
```
на примере других сервисов, Async сохраняется при Create
Это необязательно, но имя сущности Request стоит убрать

c23p1 replied:
Возвращать int не нужно

c23p1 commented on src/Application/Interfaces/Repositories/Admin/IClubCreationRequestsRepository.cs:
```diff
@@ -0,0 +1,10 @@
+public interface IClubCreationRequestsRepository
+{
+    public Task<ClubCreationRequest> GetByIdAsync(int id);
+    public Task ChangeActivityStatusAsync(ClubCreationRequest clubCreationRequest);
```
Репозиторий содержит только простые операции (Create, Read, Update, Delete), бизнес логика содержится в сервисах
В данном случае должен быть метод UpdateAsync, который принимает уже изменённую в сервисе сущность

c23p1 commented on src/Infrastructure/Persistence/Repositories/Admin/ClubCreationRequestsRepository.cs:
```diff
@@ -0,0 +1,36 @@
+    
+    public async Task CreateAsync(ClubCreationRequest clubCreationRequest)
+    {
+        clubCreationRequest.IsActive = true;
```
IsActive имеет значение по - умолчанию true, можно задать это в модели или в сервисе

c23p1 commented on src/Application/DTO/ClubCreationRequestDTO.cs:
```diff
@@ -0,0 +1,32 @@
+    public int PersonId { get; set; }
+    public int? BannerId { get; set; }
+    [MaxLength(1000)]
+    public string Comment { get; set; }
```
Required

c23p1 commented on Admin.API/Controllers/ClubCreationRequestsController.cs:
```diff
@@ -0,0 +1,54 @@
+namespace Admin.API.Controllers;
+
+[ApiController]
+[Route("creation-requests")]
```
не факт, что стандарт есть, но возможно более читаемо будет в camelCase

c23p1 commented on Admin.API/Controllers/ClubCreationRequestsController.cs:
```diff
@@ -0,0 +1,54 @@
+
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [HttpPost("changeRequestStatusById/{requestId:int}")]
+    public async Task<ActionResult> ChangeRequestStatusById(int requestId, bool status)
```
Нет ручки на AcceptRequest и DeclineRequest, изменение статуса происходит автоматически и работает только в одну сторону, нельзя сделать неактивную заявку активной, что имплементировано в сервисе
См. тз

c23p1 commented on src/Application/Services/ClubCreationRequestsService.cs:
```diff
@@ -0,0 +1,50 @@
+    {
+        var clubCreationRequest = await _clubCreationRequestsRepository.GetByIdAsync(requestId);
+        if (clubCreationRequest is null) throw new NotFoundException(nameof(ClubCreationRequest), requestId);
+        if (status)
```
Status неявно обозначает, принята ли заявка или нет, в таком случае метод должен называться HandleRequestStatus, а переменная - isAccepted. Если переменная - true, заявка должна быть помечена как неактивная.

В данном случае при status = true заявка меняет свой статус активности на противоположный, т.е. при указании true и Id закрытой заявки, клуб заявки будет создан, а заявка - переоткрыта, это баг

c23p1 replied:
Желательно, как указано в комментарии к интерфейсу, добавить 2 различных метода, тк при использовании одного происходит ограничение в виде bool isAccepted. При слиянии (временном) методов на принятие и отклонении заявки, необходим Enum статусов заявки для чистоты API

c23p1 commented on src/Application/Interfaces/Services/IClubCreationRequestsService.cs:
```diff
@@ -0,0 +1,9 @@
+public interface IClubCreationRequestsService
+{
+    public Task<int> CreateRequest(ClubCreationRequestDTO clubCreationRequestDTO);
+    public Task ChangeRequestStatus(bool status, int requestId);
```
Лучше создать два метода, как указано в тз, так как логика обработки принятия и отклонения заявки может различаться

jbkkkkk committed:
[SN-121]
Внесены изменения согласно комментариям к предыдущим коммитам

c23p1 commented on Admin.API/Controllers/ClubCreationRequestsController.cs:
```diff
@@ -0,0 +1,62 @@
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [HttpPost("createRequest")]
+    public async Task<ActionResult<ClubCreationRequestDetailsDTO>> CreateRequest(
```
Неверный return type

c23p1 commented on Admin.API/Controllers/ClubCreationRequestsController.cs:
```diff
@@ -0,0 +1,62 @@
+
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [HttpPost("acceptById/{requestId:int}")]
+    public async Task<ActionResult> AcceptRequestById(int requestId)
```
IActionResult для ответов без содержания

c23p1 commented on Admin.API/Controllers/ClubCreationRequestsController.cs:
```diff
@@ -0,0 +1,62 @@
+    
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [HttpPost("declineById/{requestId:int}")]
+    public async Task<ActionResult> DeclineRequestById(int requestId)
```
IActionResult

c23p1 commented on src/Application/DTO/ClubCreationRequestDTO.cs:
```diff
@@ -0,0 +1,33 @@
+    public string Comment { get; set; }
+}
+
+public record ClubCreationRequestDetailsDTO
```
Для деталей тоже нужны атрибуты Required для генерации моделей фронту

jbkkkkk committed:
[SN-121]
Добавлен Required для полей в ClubCreationRequestDetailsDTO. ActionResult заменен на IActionResult для ответов без содержания.

c23p1 committed:
Merge remote-tracking branch 'origin/master' into features/add-admin-panel

# Conflicts:
#	src/Infrastructure/Persistence/Data/ApplicationDbContext.cs

c23p1 committed:
Merge remote-tracking branch 'origin/master' into features/add-admin-panel

jbkkkkk committed:
[SN-121]
Добавлена зависимая сущность PersonSummaryDTO в ClubCreationRequestDetailsDTO

jbkkkkk committed:
[SN-121]
Добавлена возможность загружать изображения к запросу на создание клуба

c23p1 committed:
[SN-121]
- Исправлена ошибка, при которой заявку можно было обработать после рассмотрения
- Исправлена регистрация сервисов
- Добавлены обработчики исключений в проект Admin.API

c23p1 committed:
Merge remote-tracking branch 'origin/features/add-admin-panel' into features/add-admin-panel

c23p1 committed:
[SN-121] Исправлен код запроса на отправку формы создания клуба

c23p1 committed:
[SN-121] Внесены изменения в модели проекта
- Свойство ImageId сущности ClubCreationRequest сделано обязательным
- Изменён тип свойства PersonId модели ClubCreationRequestDTO на int?
- Добавлены необходимые атрибуты Required

c23p1 committed:
[SN-121] Изменён путь к контроллеру и методу для создания запросов на создание клуба

c23p1 committed:
[SN-121] Внесены незначительные изменения

c23p1 committed:
[SN-121] Внесены незначительные изменения
